versioned/spi/src/main/java/org/projectnessie/versioned/StoreWorker.java

```diff
@@ -23,20 +28,42 @@
  */
 public interface StoreWorker<VALUE, COMMIT_METADATA> {
 
-  ValueWorker<VALUE> getValueWorker();
+  ValueWorker<WithEntityType<VALUE>> getValueWorker();
 
   Serializer<COMMIT_METADATA> getMetadataSerializer();
 
   /**
-   * Create StoreWorker for provided helpers.
+   * Create StoreWorker for provided helpers. Adding single byte data type as first byte in serialized value.
    */
   static <VALUE, COMMIT_METADATA> StoreWorker<VALUE, COMMIT_METADATA>
       of(ValueWorker<VALUE> valueWorker, Serializer<COMMIT_METADATA> commitSerializer) {
     return new StoreWorker<VALUE, COMMIT_METADATA>() {
 
       @Override
-      public ValueWorker<VALUE> getValueWorker() {
-        return valueWorker;
+      public ValueWorker<WithEntityType<VALUE>> getValueWorker() {
```

Comment:
This is only used in tests currently and I think it shoudl mvoe into a test package. It wraps a string worker to also record the type